File: company-implementation/src/main/java/telran/employees/SalesPerson.java
package telran.employees;

import org.json.JSONObject;

public class SalesPerson extends WageEmployee{
    private float percent;
    private long sales; 
    public SalesPerson(){}
    public SalesPerson(long id, int basicSalary, String department, int wage, int hours,
     float percent, long sales) {
        super(id, basicSalary, department, wage, hours);
        this.percent = percent;
        this.sales = sales;
     }
     @Override
     public int computeSalary() {
        return (int) (super.computeSalary() + sales * percent / 100);
     }
     @Override
     protected void fillJSON(JSONObject jsonObj) {
      super.fillJSON(jsonObj);
      jsonObj.put("percent", percent);
      jsonObj.put("sales", sales);

     }
     @Override
     protected void setObject(JSONObject jsonObj) {
         super.setObject(jsonObj);
         percent = jsonObj.getFloat("percent");
         sales = jsonObj.getLong("sales");
      }
}

File: Node-Project/company/src/dto/SalesPerson.mjs
import WageEmployee from './WageEmployee.mjs';

export default class SalesPerson extends WageEmployee {
    constructor(id, basicSalary, department, wage, hours, percent, sales) {
        super(id, basicSalary, department, wage, hours);
        this.percent = percent;
        this.sales = sales;
    }

    computeSalary() {
        return super.computeSalary() + this.sales * this.percent / 100;
    }

    static fromJSON(json) {//not static...this is string
        return new SalesPerson(
            json.id, json.basicSalary, json.department,
            json.wage, json.hours, json.percent, json.sales
        );
    }

    toJSON() {
        const json = super.toJSON();
        return {
            ...json,
            percent: this.percent,
            sales: this.sales
        };
    }
}
